src/org/hammerhead226/sharkmacro/motionprofiles/ProfileParser.java

```diff
@@ -98,9 +81,32 @@ public Profile toObject(CANTalon leftTalon, CANTalon rightTalon) {
 		}
 
 		Profile p = new Profile(left, right, leftTalon, rightTalon);
-		putInCache(p);
 
 		return p;
 	}
 
+	/**
+	 * This method generates a new filename to be used for saving a new file. For
+	 * example, if the newest file in the storage directory is
+	 * {@code prefix0003.csv}, the method will return {@code prefix0004.csv}.
+	 * 
+	 * @return a new complete filename in the prefix + number naming convention
+	 */
+	public static String getNewFilename() {
+		return Constants.PROFILE_DEFAULT_PREFIX + String.format("%04d",
+				findLatestNumberedFile(Constants.PROFILE_STORAGE_DIRECTORY, Constants.PROFILE_DEFAULT_PREFIX) + 1)
+				+ ".csv";
+	}
+
+	/**
+	 * This method finds the newest file named with prefix + number naming
+	 * convention in the storage directory.
+	 * 
+	 * @return the complete filename of the latest (highest numbered) file in the
+	 *         storage directory
+	 */
+	public static String getNewestFilename() {
+		return Constants.PROFILE_DEFAULT_PREFIX + String.format("%04d",
+				findLatestNumberedFile(Constants.PROFILE_STORAGE_DIRECTORY, Constants.PROFILE_DEFAULT_PREFIX));
+	}
```

Comment:
Shouldn't this call getNewestFilename(Constants.PROFILE_STORAGE_DIRECTORY, Constants.PROFILE_DEFAULT_PREFIX)?